pkg/models/setting.go

```diff
@@ -27,13 +36,42 @@ func (s *AWSSiteWiseDataSourceSetting) Load(config backend.DataSourceInstanceSet
 		s.Profile = config.Database // legacy support (only for cloudwatch?)
 	}
 
+	// Make sure to set an auth mode
+	if s.Region == EDGE_REGION {
+		if s.EdgeAuthMode == "" {
+			s.EdgeAuthMode = EDGE_AUTH_MODE_DEFAULT
+		}
+	}
+
 	s.AccessKey = config.DecryptedSecureJSONData["accessKey"]
 	s.SecretKey = config.DecryptedSecureJSONData["secretKey"]
 	s.Cert = config.DecryptedSecureJSONData["cert"]
+	s.EdgeAuthPass = config.DecryptedSecureJSONData["edgeAuthPass"]
+	return nil
+}
+
+func (s *AWSSiteWiseDataSourceSetting) Validate() error {
+	if s.Region == EDGE_REGION {
```

Comment:
nit: check `s.region != EDGE_REGION` first, leave early rule :)

https://stackoverflow.com/questions/53115008/why-is-good-to-check-for-false-first-in-an-if-statement

Reply:
thanks for nice tidbit!